synapsecns/sdk
# Massive refactors of tests and some other things

project:
  - use node 17

`tests`:
  - all:
    - implement wrapExpect helper function: Eliminates most instances of having to type out `function(this: Mocha.Context)` for test cases.
    - a few helper functions can take messages
    - Consolidate various helper functions and constants into test/helpers
    - Use more awaits
    - reflect ChainId enum refactor
    - use function(this: Mocha.Context) instead of anonymous arrow function

  - `entities`: use test case interface
  - `slippage`: Use a test case interface for formatSlippageToString tests
  - `Basic-test.ts`:
    - small cleanup and deduplication of expect functions
    - use function(this: Mocha.Context) Mocha prefers it to arrow functions.
    - use const string[] objects for some test cases

  - `Env-test.ts`:
    - change anonymous function to function(this: Mocha.Context)
    - Use constant for test title

  - `SynapseBridge-test.ts`: rethink how promise fulfillments are tested against

`common`:
  - `chainid`: turn `ChainId.ChainId` namespace into a single enum
  - `networks`: fix `Network.supportsToken` for a few tokens
  - `synapse_contracts.ts`: add `contractsForChainId` function

`token`:
  - cleanup `WrappedToken` constructor
  - Export arguments object for `BaseToken` and `WrappedToken` as interfaces: `BaseTokenArgs`, and `WrappedTokenArgs` (respectively)
  - Change `BaseToken.valueToWei()` arg name from `amt` to `ether`
  - Remove potentially unnecessary BigNumber cast in `BaseToken.valueToWei()`

`bridge`:
  - Refactor quite a few (sometimes nested) if/else blocks into (sometimes nested) switch blocks for slightly easier readability.
  - cleanup type imports
  - `gasutils.ts`: 
    - refactor singular instances of `GasParams` into one `ChainIdTypeMap` object for consistency.
    - Aurora unfortunately needs gas now :(
  - `bridge.ts` refactor token checks to use new id field on Tokens

`tokens`: Change `isMintBurnToken()` to inline const function

`tokenswap`: use constant PoolConfig contract instance

`internal/entity`: Let's make everything an "Entity" with a unique ID
